internal/route/api/v1/repo/tree.go

```diff
@@ -0,0 +1,78 @@
+package repo
+
+import (
+	"fmt"
+
+	"github.com/gogs/git-module"
+
+	"gogs.io/gogs/internal/context"
+	"gogs.io/gogs/internal/db"
+)
+
+type repoGitTreeResponse struct {
+	Sha  string         `json:"sha"`
+	URL  string         `json:"url"`
+	Tree []*repoGitTree `json:"tree"`
+}
+
+type repoGitTree struct {
+	Path string `json:"path"`
+	Mode string `json:"mode"`
+	Type string `json:"type"`
+	Size int64  `json:"size,omitempty"`
+	Sha  string `json:"sha"`
+	URL  string `json:"url"`
+}
+
+func GetRepoGitTree(c *context.APIContext) {
+	repoPath := db.RepoPath(c.Params(":username"), c.Params(":reponame"))
+	gitTree, err := c.Repo.GitRepo.GetTree(c.Params(":sha"))
+	if err != nil {
+		c.NotFoundOrServerError("GetRepoTree", git.IsErrNotExist, err)
+		return
+	}
+	entries, err := gitTree.ListEntries()
+	if err != nil {
+		c.ServerError("GetRepoTree", err)
+		return
+	}
+	var children []*repoGitTree
+	var mode string
```

Comment:
We don't use these two variables until after the `if entries == nil {`, move them down to the their use cases is better.